### DISCUSSION TITLE
Add the ability to stage/unstage lines from all unstaged/staged hunks based on regex.

### DISCUSSION THREAD

**Discussion by isadon** [magit/magit#5219] [Ideas]
As the title states: it would be amazing if Magit allowed me to pass a regex after which all matching lines will be **_unstaged_** for already staged hunks or **_staged_** for already unstaged hunks. 

As for use case: In iOS development, and in others I'm sure, we have to deal with movement or creation of certain files in our Xcode Project causing our `project.pbxproj` file to have many changes. One such movement or creation of a file in our Xcode Project can cause 10 to 100s of line changes in the pbxproj file which produce various hunks. Those hunks then have to be individually inspected because often they contain lines which should not go into the commit being made or must be split away into another commit. We must deal with the "management" of the lines of this file whether its for changes made by us for commits that are new, or for  merges or rebases and it’s a major pain. Having the above feature would greatly help with this menial task :/. 

For some more info on the pbxproj file: https://tonisucic.medium.com/how-to-deal-with-project-pbxproj-in-conjunction-with-git-15a76fa7b5c9.

**Comment by tarsius** [comment]
While I can see how that could be useful, but I don't think this is something that should be added to Magit itself.

This probably doesn't need tight integration into Magit, so it could be implemented in a separate package your your init file.

**Reply by isadon** [reply]
Hi @tarsius I think that’s fair but can you give a bit of rationale as to why this wouldn't be a good idea to add to magit even as a one off command that can be run via `M-x magit-<something>` 😄? Do you think the use case is just a bit too obscure?

**Reply by tarsius** [reply]
Yes, I think the use case is a bit too obscure, and I do have to draw the line somewhere. Another problem is that while implementing a minimal viable version might not be too difficult, I do not want to be on the hook to implement support for all the edge-cases I expect users will run into over time.

**Reply by isadon** [reply]
Got it. Thank you!
